0.X/docker-entrypoint.sh

```diff
@@ -53,46 +58,55 @@ fi
 if [ "${1:0:1}" = '-' ]; then
-    set -- consul "$@"
+  set -- consul "$@"
 fi
 
 # Look for Consul subcommands.
 if [ "$1" = 'agent' ]; then
-    shift
-    set -- consul agent \
-        -data-dir="$CONSUL_DATA_DIR" \
-        -config-dir="$CONSUL_CONFIG_DIR" \
-        $CONSUL_BIND \
-        $CONSUL_CLIENT \
-        "$@"
+  shift
+  set -- consul agent \
+    -data-dir="$CONSUL_DATA_DIR" \
+    -config-dir="$CONSUL_CONFIG_DIR" \
+    $CONSUL_BIND \
+    $CONSUL_CLIENT \
+    "$@"
 elif [ "$1" = 'version' ]; then
-    # This needs a special case because there's no help output.
-    set -- consul "$@"
+  # This needs a special case because there's no help output.
+  set -- consul "$@"
 elif consul --help "$1" 2>&1 | grep -q "consul $1"; then
-    # We can't use the return code to check for the existence of a subcommand, so
-    # we have to use grep to look for a pattern in the help output.
-    set -- consul "$@"
+  # We can't use the return code to check for the existence of a subcommand, so
+  # we have to use grep to look for a pattern in the help output.
+  set -- consul "$@"
 fi
 
 # If we are running Consul, make sure it executes as the proper user.
 if [ "$1" = 'consul' -a -z "${CONSUL_DISABLE_PERM_MGMT+x}" ]; then
-    # If the data or config dirs are bind mounted then chown them.
-    # Note: This checks for root ownership as that's the most common case.
-    if [ "$(stat -c %u "$CONSUL_DATA_DIR")" != "$(id -u consul)" ]; then
-        chown consul:consul "$CONSUL_DATA_DIR"
-    fi
-    if [ "$(stat -c %u "$CONSUL_CONFIG_DIR")" != "$(id -u consul)" ]; then
-        chown consul:consul "$CONSUL_CONFIG_DIR"
-    fi
-
-    # If requested, set the capability to bind to privileged ports before
-    # we drop to the non-root user. Note that this doesn't work with all
-    # storage drivers (it won't work with AUFS).
-    if [ ! -z ${CONSUL_ALLOW_PRIVILEGED_PORTS+x} ]; then
-        setcap "cap_net_bind_service=+ep" /bin/consul
-    fi
-
-    set -- su-exec consul:consul "$@"
+  # Allow to setup user and group via envrironment
+  if [ -z "$CONSUL_UID" ]; then
+    CONSUL_UID="$(id -u consul)"
+  fi
+
+  if [ -z "$CONSUL_GID" ]; then
+    CONSUL_GID="$(id -g consul)"
+  fi
+
+  # If the data or config dirs are bind mounted then chown them.
+  # Note: This checks for root ownership as that's the most common case.
+  if [ "$(stat -c %u "$CONSUL_DATA_DIR")" != "${CONSUL_UID}" ]; then
+    chown ${CONSUL_UID}:${CONSUL_GID} "$CONSUL_DATA_DIR"
+  fi
+  if [ "$(stat -c %u "$CONSUL_CONFIG_DIR")" != "${CONSUL_UID}" ]; then
+    chown ${CONSUL_UID}:${CONSUL_GID} "$CONSUL_CONFIG_DIR"
+  fi
+
+  # If requested, set the capability to bind to privileged ports before
+  # we drop to the non-root user. Note that this doesn't work with all
+  # storage drivers (it won't work with AUFS).
+  if [ ! -z ${CONSUL_ALLOW_PRIVILEGED_PORTS+x} ]; then
+    setcap "cap_net_bind_service=+ep" /bin/consul
+  fi
+
+  set -- su-exec ${CONSUL_UID}:${CONSUL_GID} "$@"
```

Comment:
Is this a built-in? I am not familiar with `su-exec`. Could you clarify if there are any hidden dependencies here?

Reply:
This is part of alpine image that used as base for consul's build. As you can see su-exec was used before this patch too.

Reply:
Got it. Looks good to me!